lib/checks.js

```diff
@@ -80,7 +82,7 @@ function typeCheck(expect) {
             return value;
         }
 
-        failArgumentCheck(arguments.callee, message ||
+        failArgumentCheck(typeCheck, message ||
```

Comment:
You don't want to start the stack trace from the typeCheck function, but from the closure returned from calling the function.